07-routing-gossip.md

```diff
@@ -441,6 +448,95 @@ Other more advanced considerations involve diversity of routes to
 of local channels.
 
+### Routing Example
+
+Consider four nodes:
+
+
+```
+   B
+  / \
+ /   \
+A     C 
+ \   /
+  \ /
+   D
+```
+
+Each node asks for the following `cltv_expiry_delta` on every channel open:
+
+1. A: 10 blocks
+2. B: 20 blocks
+3. C: 30 blocks
+4. D: 40 blocks
+
+Also, each node has the same fee scheme which it uses for each of its
+channels:
+
+1. A: 100 base + 1000 millionths
+1. B: 200 base + 2000 millionths
+1. C: 300 base + 3000 millionths
+1. D: 400 base + 4000 millionths
+
+The network will see eight `channel_update` messages:
+
+1. A->B: `cltv_expiry_delta` = 20, `fee_base_msat` = 100, `fee_proportional_millionths` = 1000
+1. A->D: `cltv_expiry_delta` = 40, `fee_base_msat` = 100, `fee_proportional_millionths` = 1000
+1. B->A: `cltv_expiry_delta` = 10, `fee_base_msat` = 200, `fee_proportional_millionths` = 2000
+1. D->A: `cltv_expiry_delta` = 10, `fee_base_msat` = 400, `fee_proportional_millionths` = 4000
+1. B->C: `cltv_expiry_delta` = 30, `fee_base_msat` = 200, `fee_proportional_millionths` = 2000
+1. D->C: `cltv_expiry_delta` = 30, `fee_base_msat` = 400, `fee_proportional_millionths` = 4000
+1. C->B: `cltv_expiry_delta` = 20, `fee_base_msat` = 300, `fee_proportional_millionths` = 3000
+1. C->D: `cltv_expiry_delta` = 40, `fee_base_msat` = 300, `fee_proportional_millionths` = 3000
+
+If B were to send 4,999,999 millisatoshi to C, it wouldn't charge
+itself a fee, but it would need to consider the `cltv_expiry_delta` in
+the B->C `channel_update` (ie. the `cltv_expiry_delta` C told B in
+`open_channel` or `accept_channel`).  We also assume it adds a "shadow
+route" to give an extra CLTV of 42.  It could also add extra cltv deltas
+at other hops, as `cltv_expiry_delta` is a minimum, but we don't here for
+simplicity:
+
+   * `amount_msat`: 4999999
+   * `cltv_expiry`: current-block-height + 30 + 42
+   * `onion_routing_packet`:
+     * `amt_to_forward` = 4999999
+     * `outgoing_cltv_value` = current-block-height + 30 + 42
+
+If A were to send an 4,999,999 millisatoshi to C via B, it needs to
+pay B the fee it specified in the B->C `channel_update`, calculated as
+per [HTLC Fees](#htlc_fees):
+
+	200 + 4999999 * 2000 / 1000000 = 10199
+
+Assume a "shadow route" CLTV of 42 again, it would set `cltv_expiry`
+to the current-block-height plus the `channel_update` amounts A->B
+(20) and B->C (30), plus 42:
+
+Thus the `update_add_htlc` message from A to B would be:
+
+   * `amount_msat`: 5010198
+   * `cltv_expiry`: current-block-height + 50 + 42
+   * `onion_routing_packet`:
+     * `amt_to_forward` = 4999999
+     * `outgoing_cltv_value` = current-block-height + 30 + 42
+
+The `update_add_htlc` from B to C would be the same as the B->C direct
+payment above.
+
+Finally, if for some reason A chose the more expensive route via D, it
+would send the following `update_add_htlc` to D:
+
+   * `amount_msat`: 5020398
+   * `cltv_expiry`: current-block-height + 70 + 42
+   * `onion_routing_packet`:
+	 * `amt_to_forward` = 4999999
```

Comment:
nit: indentation

Reply:
It's nested correctly AFAICT?

Reply:
It's indented with a tab instead of spaces, which makes it appear misaligned with the next field on GitHub.